blosc/frame.c

```diff
@@ -1139,102 +1142,47 @@ int32_t frame_get_usermeta(blosc2_frame* frame, uint8_t** usermeta) {
   return usermeta_len;
 }
 
-
-int frame_get_metalayers(blosc2_frame* frame, blosc2_schunk* schunk) {
-  int32_t header_len;
-  int64_t frame_len;
-  int64_t frame_pos = FRAME_IDX_SIZE;
-  int64_t nbytes;
-  int64_t cbytes;
-  int32_t chunksize;
-  int32_t nchunks;
-  int ret = get_header_info(frame, &header_len, &frame_len, &nbytes, &cbytes, &chunksize, &nchunks,
-                            NULL, NULL, NULL, NULL, NULL);
-  if (ret < 0) {
-    BLOSC_TRACE_ERROR("Unable to get the header info from frame.");
-    return -1;
-  }
-
-  // Get the header
-  uint8_t* header = NULL;
-  if (frame->sdata != NULL) {
-    header = frame->sdata;
-  } else {
-    size_t rbytes = 0;
-    header = malloc(header_len);
-    FILE* fp = NULL;
-    if (frame->eframe) {
-      char* eframe_name = malloc(strlen(frame->urlpath) + strlen("/chunks.b2frame") + 1);
-      sprintf(eframe_name, "%s/chunks.b2frame", frame->urlpath);
-      fp = fopen(eframe_name, "rb");
-      free(eframe_name);
-    }
-    else {
-      fp = fopen(frame->urlpath, "rb");
-    }
-    if (fp != NULL) {
-      rbytes = fread(header, 1, header_len, fp);
-      fclose(fp);
-    }
-    if (rbytes != (size_t) header_len) {
-      BLOSC_TRACE_ERROR("Cannot access the header out of the fileframe.");
-      free(header);
-      return -2;
-    }
-  }
+static int frame_get_metalayers_from_header(blosc2_frame* frame, blosc2_schunk* schunk, uint8_t* header,
+  int32_t header_len) {
```

Comment:
Watch out indentation. Our style (undocumented) is to align parameters with previous line.